source/plugin/data/catalog-types.rst

```diff
@@ -0,0 +1,64 @@
+=============
+Catalog Types
+=============
+
+.. javadoc-import::
+
+    org.spongepowered.api.CatalogType
+    org.spongepowered.api.CatalogTypes
+    org.spongepowered.api.GameRegistry
+    org.spongepowered.api.text.translation.Translatable
+    org.spongepowered.api.util.annotation.CatalogedBy
+
+A big part of the data API consists of special values with a specific meaning. In previous versions of minecraft this
+have been numbers such as ``1`` that meant something specific in a given context for example a simple stone in
+the ``/give`` command. From version to version minecraft moved away from these magics numbers and moved to a more
+declarative way using (namespaced) identifiers such as ``minecraft:stone``. This however brought up two big issues with
+these values:
+
+1. How do I persist them and reload them?
+2. Which values do exist?
+
+Because of this Sponge introduced a new class of data: :javadoc:`CatalogType`\s.
+
+``CatalogType``\s provide two methods make them very versatile.
+
+First the :javadoc:`CatalogType#getId()` method that provides an easy access to a unique identifier (within that type of
+data) that can be used to save it to a config file and retrieve it back using the 
+:javadoc:`GameRegistry#getType(Class, String)` method. In fact you don't have to do this yourself, you can rely on
+Sponge's :doc:`config serialization <../configuration/serialization>` to do this for you instead. Some ids are not
```

Comment:
If this is the second method, it probably justifies a second paragraph starting at "Some ids are ..."